### PR TITLE
add youtubedl_material role

### PR DESCRIPTION
**Pull request by hereisderek**
# Description

YoutubeDL-Material is a Material Design frontend for [youtube-dl](https://rg3.github.io/youtube-dl/). It's coded using [Angular 15](https://angular.io/) for the frontend, and [Node.js](https://nodejs.org/) on the backend.

Project page: [link](https://github.com/Tzahi12345/YoutubeDL-Material)

TODO:
- [ ] doc page

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.  You can use the checkboxes below or delete them as you wish.

- [x] tested with `sb install youtubedl_material`
- [x] access the subdomain `https://ytdl.domain.com` and tested it's working

### CODE REVIEW

File: roles/youtubedl_material/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,157 @@
+##########################################################################
+# Title:            YoutubeDL-Material                                   #
+# Author(s):        Tzahi12345                                           #
```
Comment (saltydk) [inline roles/youtubedl_material/defaults/main.yml:3]: Wrong author, this is you not the upstream project creator

File: roles/youtubedl_material/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+# URL:              https://github.com/Tzahi12345/YoutubeDL-Material     #
```
Comment (saltydk) [inline roles/youtubedl_material/defaults/main.yml:4]: Supposed to be Sandbox url

File: roles/youtubedl_material/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+youtubedl_material_name: ytdl
+
+################################
+# Paths
+################################
+
+youtubedl_material_paths_folder: "{{ youtubedl_material_name }}"
+youtubedl_material_paths_location: "{{ server_appdata_path }}/{{ youtubedl_material_paths_folder }}"
+youtubedl_material_download_folder: "/mnt/unionfs/Media/{{ youtubedl_material_name }}"
```
Comment (saltydk) [inline roles/youtubedl_material/defaults/main.yml:22]: Downloads generally do not go into Media folders

File: roles/youtubedl_material/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+
+################################
+# Web
+################################
+
+youtubedl_material_web_subdomain: "{{ youtubedl_material_name }}"
+youtubedl_material_web_domain: "{{ user.domain }}"
+youtubedl_material_web_port: "17442"
+youtubedl_material_web_url: "{{ 'https://' + youtubedl_material_web_subdomain + '.' + youtubedl_material_web_domain
```
Comment (saltydk) [inline roles/youtubedl_material/defaults/main.yml:40]: Look at other roles for a reference on how to format this

File: roles/youtubedl_material/defaults/main.yml [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+youtubedl_material_dns_record: "{{ youtubedl_material_web_subdomain }}"
+youtubedl_material_dns_zone: "{{ youtubedl_material_web_domain }}"
+youtubedl_material_dns_proxy: "{{ dns.proxied }}"
+
+################################
+# Traefik
+################################
+
+youtubedl_material_traefik_sso_middleware: "{{ traefik_default_sso_middleware }}"
+youtubedl_material_traefik_middleware: "{{ traefik_default_middleware + ',' + youtubedl_material_traefik_sso_middleware
```
Comment (saltydk) [inline roles/youtubedl_material/defaults/main.yml:57]: Look at other roles for a reference on how to format this

File: roles/youtubedl_material/defaults/main.yml [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+# Image
+youtubedl_material_docker_image_pull: true
+youtubedl_material_docker_image_tag: "latest"
+youtubedl_material_docker_image: "tzahi12345/youtubedl-material:{{ youtubedl_material_docker_image_tag }}"
+
+# Ports
+youtubedl_material_docker_ports_defaults: []
+youtubedl_material_docker_ports_custom: []
+youtubedl_material_docker_ports: "{{ youtubedl_material_docker_ports_defaults
```
Comment (saltydk) [inline roles/youtubedl_material/defaults/main.yml:78]: Look at other roles for a reference on how to format this

File: roles/youtubedl_material/defaults/main.yml [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+# Envs
+youtubedl_material_docker_envs_default:
+  PUID: "{{ uid }}"
+  PGID: "{{ gid }}"
+  TZ: "{{ tz }}"
+  # ytdl_use_local_db: 'false'
+  write_ytdl_config: 'true'
+youtubedl_material_docker_envs_custom: {}
+youtubedl_material_docker_envs: "{{ youtubedl_material_docker_envs_default
```
Comment (saltydk) [inline roles/youtubedl_material/defaults/main.yml:89]: Look at other roles for a reference on how to format this

File: roles/youtubedl_material/defaults/main.yml [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+youtubedl_material_docker_ports_custom: []
+youtubedl_material_docker_ports: "{{ youtubedl_material_docker_ports_defaults
+                             + youtubedl_material_docker_ports_custom }}"
+
+# Envs
+youtubedl_material_docker_envs_default:
+  PUID: "{{ uid }}"
+  PGID: "{{ gid }}"
+  TZ: "{{ tz }}"
+  # ytdl_use_local_db: 'false'
```
Comment (saltydk) [inline roles/youtubedl_material/defaults/main.yml:86]: This is useless, either implement a variable to toggle this behavior or remove it

File: roles/youtubedl_material/defaults/main.yml [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+
+# Ports
+youtubedl_material_docker_ports_defaults: []
+youtubedl_material_docker_ports_custom: []
+youtubedl_material_docker_ports: "{{ youtubedl_material_docker_ports_defaults
+                             + youtubedl_material_docker_ports_custom }}"
+
+# Envs
+youtubedl_material_docker_envs_default:
+  PUID: "{{ uid }}"
```
Comment (saltydk) [inline roles/youtubedl_material/defaults/main.yml:83]: Based on a quick glance the container does not support PUID/PGID check [here](https://github.com/Tzahi12345/YoutubeDL-Material#custom-uidgid)

File: roles/youtubedl_material/defaults/main.yml [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+  # ytdl_use_local_db: 'false'
+  write_ytdl_config: 'true'
+youtubedl_material_docker_envs_custom: {}
+youtubedl_material_docker_envs: "{{ youtubedl_material_docker_envs_default
+                            | combine(youtubedl_material_docker_envs_custom) }}"
+
+# Commands
+youtubedl_material_docker_commands_default: []
+youtubedl_material_docker_commands_custom: []
+youtubedl_material_docker_commands: "{{ youtubedl_material_docker_commands_default
```
Comment (saltydk) [inline roles/youtubedl_material/defaults/main.yml:95]: Look at other roles for a reference on how to format this

File: roles/youtubedl_material/defaults/main.yml [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+
+# Volumes
+youtubedl_material_docker_volumes_default:
+  - "{{ youtubedl_material_paths_location }}/appdata:/app/appdata"
+  - "{{ youtubedl_material_paths_location }}/users:/app/users"
+  - "{{ youtubedl_material_download_folder }}/audio:/app/audio"
+  - "{{ youtubedl_material_download_folder }}/video:/app/video"
+  - "{{ youtubedl_material_download_folder }}/subscriptions:/app/subscriptions"
+youtubedl_material_docker_volumes_custom: []
+youtubedl_material_docker_volumes: "{{ youtubedl_material_docker_volumes_default
```
Comment (saltydk) [inline roles/youtubedl_material/defaults/main.yml:106]: Look at other roles for a reference on how to format this and remove docker_volumes_downloads_common as it is already included

File: roles/youtubedl_material/defaults/main.yml [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+                               + youtubedl_material_docker_volumes_custom
+                               + docker_volumes_downloads_common }}"
+
+# Devices
+youtubedl_material_docker_devices_default: []
+youtubedl_material_docker_devices_custom: []
+youtubedl_material_docker_devices: "{{ youtubedl_material_docker_devices_default
```
Comment (saltydk) [inline roles/youtubedl_material/defaults/main.yml:113]: Look at other roles for a reference on how to format this

File: roles/youtubedl_material/defaults/main.yml [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+                               + youtubedl_material_docker_devices_custom }}"
+
+# Hosts
+youtubedl_material_docker_hosts_default: []
+youtubedl_material_docker_hosts_custom: []
+youtubedl_material_docker_hosts: "{{ docker_hosts_common
```
Comment (saltydk) [inline roles/youtubedl_material/defaults/main.yml:119]: Look at other roles for a reference on how to format this

File: roles/youtubedl_material/defaults/main.yml [hunk 14]
```diff
@@ -0,0 +1,157 @@
...
+                             | combine(youtubedl_material_docker_hosts_default)
+                             | combine(youtubedl_material_docker_hosts_custom) }}"
+
+# Labels
+youtubedl_material_docker_labels_default: {}
+youtubedl_material_docker_labels_custom: {}
+youtubedl_material_docker_labels: "{{ docker_labels_common
```
Comment (saltydk) [inline roles/youtubedl_material/defaults/main.yml:126]: Look at other roles for a reference on how to format this

File: roles/youtubedl_material/defaults/main.yml [hunk 15]
```diff
@@ -0,0 +1,157 @@
...
+                              | combine(youtubedl_material_docker_labels_custom) }}"
+
+# Hostname
+youtubedl_material_docker_hostname: "{{ youtubedl_material_name }}"
+
+# Networks
+youtubedl_material_docker_networks_alias: "{{ youtubedl_material_name }}"
+youtubedl_material_docker_networks_default: []
+youtubedl_material_docker_networks_custom: []
+youtubedl_material_docker_networks: "{{ docker_networks_common
```
Comment (saltydk) [inline roles/youtubedl_material/defaults/main.yml:137]: Look at other roles for a reference on how to format this

File: roles/youtubedl_material/defaults/main.yml [hunk 16]
```diff
@@ -0,0 +1,157 @@
...
+                                + youtubedl_material_docker_networks_default
+                                + youtubedl_material_docker_networks_custom }}"
+
+# Capabilities
+youtubedl_material_docker_capabilities_default: []
+youtubedl_material_docker_capabilities_custom: []
+youtubedl_material_docker_capabilities: "{{ youtubedl_material_docker_capabilities_default
```
Comment (saltydk) [inline roles/youtubedl_material/defaults/main.yml:144]: Look at other roles for a reference on how to format this

File: roles/youtubedl_material/defaults/main.yml [hunk 17]
```diff
@@ -0,0 +1,157 @@
...
+                                    + youtubedl_material_docker_capabilities_custom }}"
+
+# Security Opts
+youtubedl_material_docker_security_opts_default: []
+youtubedl_material_docker_security_opts_custom: []
+youtubedl_material_docker_security_opts: "{{ youtubedl_material_docker_security_opts_default
```
Comment (saltydk) [inline roles/youtubedl_material/defaults/main.yml:150]: Look at other roles for a reference on how to format this

File: sandbox.yml [hunk 18]
```diff
@@ -182,5 +182,6 @@
     - { role: xteve, tags: ['xteve'] }
     - { role: yacht, tags: ['yacht'] }
     - { role: your_spotify, tags: ['your-spotify'] }
+    - { role: youtubedl_material, tags: ['youtubedl_material'] }
```
Comment (saltydk) [inline sandbox.yml:185]: We no longer accept underscores in tags

**Review by saltydk — CHANGES_REQUESTED** [review]
Please review the requested changes.

**Comment by saltydk** [comment]
Close this if you don't want to do the changes for him @owine since he hasn't responded for 2 weeks I assume this will remain stagnant.

**Comment by owine** [comment]
Happy to reopen and revisit if the requested changes are made

**Pull request closed without merging**